### DISCUSSION TITLE
Miden Blockexplorer Feedback

### DISCUSSION THREAD

**Discussion by Dominik1999** [0xPolygonMiden/examples#192] [General]
Post your feedback, feature requests or general ideas here in the thread!

**Comment by polydez** [comment]
It would be nice if the block explorer updated counters in real time. Now we have to update the page manually. And this also would be cool to have automatic updates for block/tx/account updates lists. One good example is here (turn on "update" switch above the message list): https://ever.live/messages

**Comment by bobbinth** [comment]
Some preliminary comments:

### Main page

- I'm not sure we need "account updates" box - this should always be the same as the number of transactions
  - Currently, these counts are off by 2 - curious why that's the case.
- I would probably rename "Notes" into "Created notes".
- It would be really nice to get some of these numbers to update dynamically (i.e., the block count, the transaction count etc.)
  - This could include a "dynamic row" in the block/transaction tables to say something "10 new blocks since refresh".
- In the selector to the right of the search field (where it says "Blockchain"), I'd have the following options:
  - Blocks
  - Accounts
  - Transactions
  - Notes

### Block page

- Overview tab
  - We don't need to show "Sub hash".
- Account updates tab
  - Instead of "committed block number" (which would be the same for all account updates in the block), we should probably show IDs of the transactions that updated the account in this block.
- I'm not sure if we actually need both "Transactions" and "Account Updates" tab - they contain kind of similar info. So, if we can make the changes to the account updates tab mentioned in the previous comment, we can probably get rid of the transactions tab.
- I would rename "Notes" tab into "Created notes".
- We also need "Consumed notes" tab which would show note nullifiers created in this block.
- Would be great to show the number of elements for each tab. For example: `Account updates (4)`, `Created Note (5)`.
- We should update the text for the hints that show up on hover for various elements (I guess we'll need to provide the exact descriptions for these).

### Transactions page

- I would change the on-hover tips for titles as follows:
  - "Account ID": "ID of the account updated by the transaction"
  - "Committed block number": "Number of the block in which the transaction was committed".

### Transaction page

- Similar changes to on-hover tips as above.
- I'm not sure there is much value in the "Block hash" field being there - so, we can probably remove it.

### Accounts page

Does not open: shows "Internal server error"

### Account page

I've reviewed only the private account page so far.

- "Storage Model" should be "Storage mode".
- We should also show the most current "State hash" of the account.
- Instead of "Transactions" tab, it may make sense to show "Updates" tab. This tab could have a table with the following fields:
  - Block number ("The block in which the update was committed")
  - State hash ("Hash of the new account state after this block")
  - Transaction ID ("Transactions executed against the account in this block") - a list of one or more transactions.
- We should also add "Notes" tab. This would show all notes sent out from the account (i.e., where `note.sender == account_id`).

### Note page

I reviewed only private notes so far.

- Note Type - possible values should be `Public` or `Private` (in the future we'll also have `Encrypted` but no need to add it for now).
  - The on-hover hint should also be updated accordingly.
- We should probably combine Batch Index and "Note Index in batch" into a single "Note index" ("Index of the node in the block").
- I don't think showing "Block hash" is adding much value here - I'd probably remove this field.
- I would change the on-hover hints for the following fields:
  -  "Note Tag": "Metadata used for note routing"
  - "Note Aux": "Arbitrary user-defined data"
  - "Block number": "Number of the block in which the note was created"

Would be really cool if we could download a note inclusion proof from this page - but I'll need to define what exactly this means.

### Consumed notes page

I think right now this shows the same data as the "Notes" page. instead, it should show show a table of nullifiers. Specifically, this table would have 2 columns: "Note nullifier" and "Committed block number".


### Search by tag

It would be great to be able to search for notes by their tag. Maybe there should be a separate "Note tags" page where we list all tags and then when we click on a specific tag, we can see all notes which have this tag.

**Reply by bobbinth** [reply]
Updated the above with some additional points. Overall, I think I went through pretty much everything except for public note/account views.
